wrapanapi/systems/openstack.py

```diff
@@ -584,7 +585,8 @@ class OpenstackSystem(System, VmMixin, TemplateMixin):
     can_pause = True
 
     def __init__(self, **kwargs):
-        self.keystone_version = kwargs.get('keystone_version', 2)
+        parsed_keystone_version = re.findall(r'v[2-3]$', kwargs['auth_url'])
```

Comment:
Making assumption here that auth_url will always have `v2/v3` if not, this will be empty. Causing IndexError. Shall we try to handle it?

Reply:
Thoughts on reworking only define `parsed_keystone_version` if the `keystone_version` is not in the kwargs, that way you can put the try/except in there.

i.e. 
```python,
keystone_version = kwargs.get("keystone_version")
if not keystone_version:
    try:
        ...
    except IndexError:
        log warning/error
```

Reply:
We definitely need to handle this indexerror here (or include `if parsed_keystone_version`), I'd log a warning that no keystone version was parsed from the auth_url and then default it to 2.

I would also just use `parsed_keystone_version[0][-1]` to use the last character of that parsed version, instead of again making assumption/default to `2` or `3` directly.

Additionally, if

Reply:
@mshriver @john-dupuy updated the PR.